### PR TITLE
Add concern which handles reordering records within a class

### PR DESCRIPTION
**Pull request by davidgisbey**
## Description

This sort of follows on from https://github.com/alphagov/whitehall/pull/8557 which adds an association extension for reordering a models associations using the various ordering columns.

This adds very similar functionality but within a concern which handles reordering records within a class .

The final couple of commits update TakePartPages' to use the new concern. This may or may not be better. Very open to discussion. I've laid out my logic for doing so in the commit message 

## Trello card

https://trello.com/c/JoE5X6uD/2205-at-least-one-workflow-in-whitehall-relies-on-skipping-validations-as-some-organisations-are-known-to-be-invalid

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

File: app/models/user_orderable_class.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+module UserOrderableClass
```
Comment (ryanb-gds) [inline app/models/user_orderable_class.rb:1]: I'd probably call this one `UserOrderable` and the extension `UserOrderableExtension` or `UserOrderableRelation` I think. Calling a module a class is potentially confusing!

I'd actually consider nesting the extension inside this module, but not too sure what the impact of that would be.

**Review by ryanb-gds — CHANGES_REQUESTED** [review]
I think I'd like for us to decouple the reorderable module tests from our application models, if at all possible.

**Comment by ryanb-gds** [comment]
> I think I'd like for us to decouple the reorderable module tests from our application models, if at all possible.

Also, and I should have led with this, great work! Love that we're pulling database logic down into the model layer where it belongs and wrapping it in a single API

File: app/models/user_orderable_class.rb (hunk 1 above)
Comment (davidgisbey) [reply app/models/user_orderable_class.rb]: cool have done. I'll give nesting the extension a go

**Comment by davidgisbey** [comment]
I think this ones ready for a re-review when you get a chance please @ryanb-gds

**Review by ryanb-gds — APPROVED** [review]
Looks great, super job 🚀

**Comment by davidgisbey** [comment]
Thanks so much for all the feedback & the review 👍

**Pull request merged**